test/tests/searchTest.js

```diff
@@ -83,4 +83,57 @@ describe("Zotero.Search", function() {
 			assert.propertyVal(conditions[0], 'value', 'foo')
 		});
 	});
+
+	describe("#search()", function () {
+		let win;
+		let fooItem;
+		let foobarItem;
+
+		before(function* () {
+			// Hidden browser, which requires a browser window, needed for charset detection
+			// (until we figure out a better way)
+			win = yield loadBrowserWindow();
+			fooItem = yield importFileAttachment("search/foo.html");
+			foobarItem = yield importFileAttachment("search/foobar.html");
+		});
+
+		after(function () {
+			if (win) {
+				win.close();
+			}
+		});
+
+		it("should return results with full-text conditions", function* () {
+			let s = new Zotero.Search();
+			s.addCondition('fulltextWord', 'contains', 'foo');
+			let results = yield s.search();
+			assert.equal(results[0], fooItem.id);
+		});
+
+		it("should not return non-matches with full-text conditions", function* () {
+			let s = new Zotero.Search();
+			s.addCondition('fulltextWord', 'contains', 'baz');
+			let results = yield s.search();
+			assert.lengthOf(results, 0);
+		});
+
+		it("should return results for full-text conditions in ALL mode", function* () {
+			let s = new Zotero.Search();
+			s.addCondition('joinMode', 'all');
+			s.addCondition('fulltextWord', 'contains', 'foo');
+			s.addCondition('fulltextWord', 'contains', 'bar');
+			let results = yield s.search();
+			assert.equal(results[0], foobarItem.id);
```

Comment:
But that's only if the members are duplicated, no? That's a good point though nonetheless.